### ISSUE TITLE
Entering a second custom tag that starts with the same text as the previous custom tag removes the previous custom tag

### ISSUE THREAD

**Issue by senseijames** [ng-select/ng-select#1033]
**Describe the bug**
Entering a second custom tag that starts with the same text as the previous custom tag will remove the previous custom tag.  E.g. you cannot create "cool tag" and then "cool" - trying to create "cool" will delete the previous tag "cool tag".

**Reproducbile example**
Custom Tags dropdown at the ng-select demo stackblitz (third example from the top)
https://stackblitz.com/edit/ng-select?file=app%2Fapp.component.ts

**To Reproduce** 
1. Enter the custom tag "tag 1"
2. Enter the custom tag "tag"
Result:
Entering the second tag “tag” deletes the first tag “tag 1”.  This is readibly reproducible with any string in place of “1”; e.g. entering “something cool” and then “something” will delete “something cool”.

**Expected behavior**
Entering "tag" after "tag 1" would create two tags: "tag 1" and "tag"

**Desktop (please complete the following information):**
 - OS: OS X, probably also Windows
 - Browser Chrome
 - Version latest

**Additional context**
100% Repro

**Comment by varnastadeus** [comment]
when you are entering same term, your last added item is highlighted and enter un-selects it. This is by design. If you want to add then focus tag item and press enter

**Comment by senseijames** [comment]
But you are not entering the same term.  Here, you are entering a term that is only _similar to the previous term_.  For example, if I have entered “cool tag” and then I want to add “cool” right after it, I haven’t re-entered the same text ‘cool tag’.  I should be able to type in “cool” there and have it added to the set.

**Comment by varnastadeus** [comment]
If you are using this component purely for tag creation, then you should check tag example without dropdown or use [hideSelected]

**Comment by senseijames** [comment]
“I see!” said the blind man

You’re 1000% correct – this is not a bug, it’s merely developer error.  `hideSelected` is exactly what we needed.

Thank you, good Sir!

(For this, and for an awesome component!)

PS - You _may_ consider updating the documentation from "Allows to hide selected items" to something like "Allows selected items to be hidden.  Note that entering a tag that starts with the same string as any previous tag will deselect those previous tags".  A bit wordy, I know, but it may help preclude dev confusion.
